### ISSUE TITLE
context rule triggers for context type aliases

### ISSUE THREAD

**Issue by trofkm** [go-simpler/sloglint#125]
Hi!

`-context-only=all` reports a diagnostic when calling ErrorContext with a type alias of context.Context (or a similar wrapper type)

Minimal repro:

```go
package main

import (
	"context"
	"log/slog"
	"os"
)

type Ctx = context.Context

func main() {
	log := slog.New(slog.NewTextHandler(os.Stdout, nil))
	var ctx Ctx
	log.ErrorContext(ctx, "test") // trigger ErrorContextContext
}
```

launch sloglint with --context-rule=only
```sloglint -context-only=all```

So the error says that:

```ErrorContextContext should be used instead```

 It seems the rule just takes the logger method name and blindly adds a Context suffix. The real problem appears to be in the logic that checks whether the argument type is a context.Context alias or similar.

**Comment by trofkm** [comment]
In my case I used https://github.com/vladopajic/go-actor with its internal type alias `actor.Context`

**Comment by tmzane** [comment]
I can't reproduce on the latest `main`. I did some refactoring recently, probably already fixed. Thanks for the report.
